jfix-stdlib-ratelimiter/src/main/kotlin/ru/fix/stdlib/ratelimiter/DispatcherWaitingScope.kt

```diff
@@ -0,0 +1,52 @@
+package ru.fix.stdlib.ratelimiter
+
+import kotlinx.coroutines.*
+import mu.KotlinLogging
+import ru.fix.dynamic.property.api.DynamicProperty
+import kotlin.coroutines.CoroutineContext
+
+open class DispatcherWaitingScope(
+        private val waitTimeout: DynamicProperty<Long>,
+        context: CoroutineContext = Dispatchers.Default
+) : CoroutineScope {
+
+    companion object {
+        val log = KotlinLogging.logger { }
+    }
+
+    private val parentJob = SupervisorJob()
+
+    override val coroutineContext: CoroutineContext = context + parentJob
+
+    /**
+     * Ожидает завершения работы корутин, но не больше, чем [waitTimeout]
+     *
+     * Вернет true, если все корутины контекста завершили свою работу
+     * иначе вернет false
+     */
+    fun waitChildrenAndCancel(): Boolean {
+        val waitingResult = runBlocking {
+            try {
+                withTimeout(waitTimeout.get()) {
+                    parentJob.children.forEach { c -> c.join() }
+                }
+            } catch (e: TimeoutCancellationException) {
+                log.warn(e) {
+                    "Scope couldn't complete execution for $waitTimeout ms. All coroutines will be " +
```

Comment:
As I saw from tests, the log doesn't write current timeout value, but `ru.fix.dynamic.property.api.ConstantProperty@345f9603 ms`


```suggestion
            val waitTimeoutMs = waitTimeout.get()
            try {
                withTimeout(waitTimeoutMs) {
                    parentJob.children.forEach { c -> c.join() }
                }
            } catch (e: TimeoutCancellationException) {
                log.warn(e) {
                    "Scope couldn't complete execution for $waitTimeoutMs ms. All coroutines will be " +
                            "canceled. Actual pending operations count: ${getPendingOperations()}"
                }
```

Reply:
Fixed. By the way, it was copy-paste from our main project

Reply:
Fixed, by the way, it was taken from our main project